PROTOCOL.md

```diff
@@ -265,19 +265,25 @@ The following is an example `metaData` action:
 <!-- TODO: forward references configuration options -->
 
 ### Add File and Remove File
-The `add` and `remove` actions are used to modify the data in a table by adding or removing individual data files respectively.
+The `add` and `remove` actions are used to modify the data in a table by adding or removing individual _logical files_ respectively.
 
-The path of a file acts as the primary key for the entry in the set of files.
-When an `add` action is encountered for a path that is already present in the table, statistics and other information from the latest version should replace that from any previous version.
-As such, additional statistics can be added for a path already present in the table by adding it again.
+Every _logical file_ of the table (referred to as just a _file_ going forward) is represented by a path to a data file, combined with a Deletion Vector (DV) that indicates which rows of the data file are no longer in the table. The path of the data file acts as the primary key for the entry in the set of files. Deletion Vectors are an optional feature. In tables that do not have this feature enabled, all Deletion Vectors are considered to be empty.
+
+When an `add` action is encountered for a file (i.e., a data file and Deletion Vector combination) that is already present in the table, statistics and other information from the latest version should replace that from any previous version.
+The primary key for the entry of a file in the set of files is a tuple of the data file's `path` as well as a unique id describing the DV.
 
 The `remove` action includes a timestamp that indicates when the removal occurred.
-Physical deletion of the file can happen lazily after some user-specified expiration time threshold.
+Physical deletion of the data and Deletion Vector file can happen lazily after some user-specified expiration time threshold.
```

Comment:
here is a case where defining "physical files" would be better. Constantly referring to DVs in this entire section make this section hard to read for the simple case no-DV case.

Reply:
Simplified this.